client/src/hooks/helpers/battles/useBattles.tsx

```diff
@@ -164,51 +138,44 @@ export const getBattleInfoByOwnArmyEntityId = (ownArmyEntityId: bigint): ExtraBa
     .values()
     .next().value;
 
-  const opponentArmy = getComponentValue(Army, opponentArmyEntityId);
+  const opponentArmy = getComponentValue(Army, opponentArmyEntityId!);
```

Comment:
Can opponentArmyEntityId be undefined if all enemy armies already left the battle ?